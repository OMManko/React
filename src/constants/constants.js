export const endPoint = 'http://react-cdp-api.herokuapp.com/movies';

export const moviePosterSizes = {
    SMALL: 'moviePoster__small',
    MEDIUM: 'moviePoster__medium'
};

export const sortOptions = {
    RELEASE: 'release',
    RATING: 'vote_average'
};

export const filterOptions = {
    TITLE: 'title',
    GENRE: 'genres'
};


export const actionTypes = {
    SEARCH_BY: 'SEARCH_BY',
    SORT_BY: 'SORT_BY',
    SEARCH_MOVIES: 'SEARCH_MOVIES',
    UPDATE_INPUT_VALUE: 'UPDATE_INPUT_VALUE',
    SET_SELECTED_MOVIE: 'SET_SELECTED_MOVIE',
    RESET_SELECTED_MOVIE: 'RESET_SELECTED_MOVIE'
};

export const initialState = {
    moviesList: [],
    moviesFound: 0,
    searchInputValue: '',
    selectedSortOption: sortOptions.RELEASE,
    selectedFilterOption: filterOptions.TITLE,
    selectedMovieInfo: {}
};
