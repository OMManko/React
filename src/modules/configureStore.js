import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware, { END } from 'redux-saga';

import movieAppReducer from '../reducers/reducers';
import { rootSaga } from '.';

const sagaMiddleware = createSagaMiddleware();

export default (initialState) => {
  const store = createStore(movieAppReducer, initialState, applyMiddleware(sagaMiddleware));

  sagaMiddleware.run(rootSaga);
  store.runSaga = () => sagaMiddleware.run(rootSaga);
  store.close = () => store.dispatch(END);

  return store;
};
