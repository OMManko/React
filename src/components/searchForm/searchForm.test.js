import { mount } from 'enzyme';
import React from 'react';
import SearchForm from '../searchForm/searchForm';
import TextField from "../shared/input/input";
import RadioGroup from "../shared/radioGroup/radioGroup";
import { Provider } from "react-redux";
import { initialState } from "../../constants/constants";
import configureStore from 'redux-mock-store';

const mockStore = configureStore();

let component;
let store;

describe('<SearchForm />', () => {
    beforeEach(() => {
        store = mockStore(initialState);
        component = mount(<Provider store={store}><SearchForm/></Provider>);
    });

    it('should render search form and match snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    it('should contain text field', () => {
        expect(component.find(TextField).length).toEqual(1);
    });

    it('should contain radio group', () => {
        expect(component.find(RadioGroup).length).toEqual(1);
    });
});
