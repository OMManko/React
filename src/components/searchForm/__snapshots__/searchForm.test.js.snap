// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> should render search form and match snapshot 1`] = `ShallowWrapper {}`;
